File: backend/flight_booking_project/src/main/java/com/javaweb/flight_booking_project/dto/response/TicketBookedInfo.java
package com.javaweb.flight_booking_project.dto.response;

import lombok.Data;

import java.time.LocalDate;
import java.util.Date;

@Data
public class TicketBookedInfo {
    private Long ticketId;
    private double price;
    private String seatNumber;
    private String flightCode;
    private Date arrivalTime;
    private Date departureTime;
    private String arrivalAirportName;
    private String departureAirportName;
    private String airlineName;
    private LocalDate createAt ;
    private String email;
    private String name;
    private String phone;
    private String nameSeat;
    private String seatClass;
    private String logo;
    private double luggage;

    public TicketBookedInfo() {}

    public TicketBookedInfo(Long ticketId, double price, String seatNumber, String flightCode, Date arrivalTime, Date departureTime, String arrivalAirportName, String departureAirportName, String airlineName) {
        this.ticketId = ticketId;
        this.price = price;
        this.seatNumber = seatNumber;
        this.flightCode = flightCode;
        this.arrivalTime = arrivalTime;
        this.departureTime = departureTime;
        this.arrivalAirportName = arrivalAirportName;
        this.departureAirportName = departureAirportName;
        this.airlineName = airlineName;
    }
}

File: mobile/lib/data/model/ticket_booked_info.dart
class TicketBookedInfo {
  int ticketId;
  double price;
  String seatNumber;
  String flightCode;
  DateTime? arrivalTime;
  DateTime? departureTime;
  String arrivalAirportName;
  String departureAirportName;
  String airlineName;
  double luggage;

  TicketBookedInfo({
    required this.ticketId,
    required this.price,
    required this.seatNumber,
    required this.flightCode,
    required this.arrivalTime,
    required this.departureTime,
    required this.arrivalAirportName,
    required this.departureAirportName,
    required this.airlineName,
    required this.luggage,
  });

  factory TicketBookedInfo.fromJson(Map<String, dynamic> json) {
    return TicketBookedInfo(
      ticketId: json["ticketId"],
      price: json["price"],
      seatNumber: json["seatNumber"],
      flightCode: json["flightCode"],
      arrivalTime: json['arrivalTime'] != null ? DateTime.parse(json['arrivalTime']) : null,
      departureTime: json['departureTime'] != null ? DateTime.parse(json['departureTime']) : null,
      arrivalAirportName: json["arrivalAirportName"],
      departureAirportName: json["departureAirportName"],
      airlineName: json["airlineName"],
      luggage: json["luggage"],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "ticketId": ticketId,
      "price": price,
      "seatNumber": seatNumber,
      "flightCode": flightCode,
      "arrivalTime": arrivalTime,
      "departureTime": departureTime,
      "arrivalAirportName": arrivalAirportName,
      "departureAirportName": departureAirportName,
      "airlineName": airlineName,
      "luggage": luggage,
    };
  }

  static List<TicketBookedInfo> fromJsonList(List<dynamic> jsonList) {
    return jsonList.map((json) => TicketBookedInfo.fromJson(json)).toList();
  }
}